gradio/oauth.py

```diff
@@ -124,19 +123,50 @@ def _add_mocked_oauth_routes(app: fastapi.FastAPI) -> None:
     @app.get("/login/huggingface")
     async def oauth_login(request: fastapi.Request):
         """Fake endpoint that redirects to HF OAuth page."""
-        return RedirectResponse("/login/callback")
+        # Define target (where to redirect after login)
+        redirect_uri = _generate_redirect_uri(request)
+        return RedirectResponse(
+            "/login/callback?" + urllib.parse.urlencode({"_target_url": redirect_uri})
+        )
 
     @app.get("/login/callback")
     async def oauth_redirect_callback(request: fastapi.Request) -> RedirectResponse:
         """Endpoint that handles the OAuth callback."""
         request.session["oauth_info"] = mocked_oauth_info
-        return RedirectResponse("/")
+        return _redirect_to_target(request)
 
     @app.get("/logout")
     async def oauth_logout(request: fastapi.Request) -> RedirectResponse:
         """Endpoint that logs out the user (e.g. delete cookie session)."""
         request.session.pop("oauth_info", None)
-        return RedirectResponse("/")
+        return _redirect_to_target(request)
```

Comment:
I haven't tested this out yet, but if I follow the logic of the code, visiting the `/logout` route won't preserve the query parameters in the URL. That's because the `_redirect_to_target()` method looks for the query parameters to be encoded as the value of the `_target_url` query parameters.

I think you can simply do something like:

```
    full_url_str = str(request.url)
    new_url = full_url_str.replace("/logout", "/")
    return RedirectResponse(url=new_url)
```

and it should keep the query parms

Reply:
Yes that would work indeed. I did not thought about redirecting with correct params after logout. I'd need to update the frontend part as well (the small js snippet). Let's merge https://github.com/gradio-app/gradio/pull/7063 first and then I'll adapt.

Reply:
#7063 is merged! => updated in https://github.com/gradio-app/gradio/pull/7034/commits/3c11ea4b0b8cb840dedcb397a1931056e1c27ada with your advice